backend/server.py

```diff
@@ -110,152 +178,182 @@ class Measurement(db.Model):
     @classmethod
     def create(cls, measurement_data, hierarchy_id, node_id):
         new_measurement=Measurement(
-            node_id=node_id,
             hierarchy_id=hierarchy_id,
+            node_id=node_id,
 
             name=measurement_data["measurementName"],
             type=measurement_data["measurementType"],
-
-                
         )
 
         db.session.add(new_measurement)
         db.session.commit() # measurement now has a unique id
 
-        return new_measurement.id
+        return new_measurement
 
     @classmethod
     def create_measurements(cls, measurements_lst, hierarchy_id, node_id):
         for measurement_data in measurements_lst:
             Measurement.create(measurement_data, hierarchy_id, node_id)
 
+    @classmethod
+    def get(cls, measurement_id):
+        return Measurement.query.filter_by(id=measurement_id).first()    
+
+    @classmethod
+    def get_all(cls, hierarchy_id, node_id):
+        return Measurement.query.filter_by(hierarchy_id=hierarchy_id, node_id=node_id)
+
+    def to_dict(self):
+        measurement_dict = {
+                "id": str(self.id),
+
+                "name": self.name,
+                "type": self.type,
+                "valueFunction": self.value_function,
+            }
+        
+        return measurement_dict
+
+    @classmethod
+    def get_list(cls, hierarchy_id, node_id):
+        measurements = Measurement.get_all(hierarchy_id, node_id)
+        measurement_list = []
+
+        for measurement in measurements:
+            measurement_list.append(measurement.to_dict())
+        
+        return measurement_list
+
 
 # routes
 @app.route("/", methods=['GET'])
 def hello_world():
     return "<p>Hello World!</p>"
 
 
+# TODO Add checks that hierarchies/nodes exist before adding subordinate resources
 @app.route("/hierarchy", methods=['POST'])
```

Comment:
Would be good if you could split up these routes into files based off of similar routes. So all routes starting with "/hierarchy" would be in the same file. Flask has blueprints for this purpose and they seem easy to use.

Reply:
I'm holding off on splitting out routes because that'll probably be part of a larger refactor.

I did figure out how to split out the models into separate files though.